clio-client/src/main/scala/org/broadinstitute/clio/client/ClioClient.scala

```diff
@@ -1,14 +1,25 @@
 package org.broadinstitute.clio.client
 
-import org.broadinstitute.clio.client.commands.CommandDispatch
-import org.broadinstitute.clio.client.parser.{BaseArgs, BaseParser}
-import org.broadinstitute.clio.client.webclient.ClioWebClient
-
 import akka.actor.ActorSystem
+import caseapp.{CommandAppWithPreCommand, RemainingArgs}
+import org.broadinstitute.clio.client.commands.CustomArgParsers._
+import org.broadinstitute.clio.client.commands.{
+  Command,
+  CommandDispatch,
+  CommonOptions
+}
+import org.broadinstitute.clio.client.webclient.ClioWebClient
 
 import scala.concurrent.ExecutionContext
 
-object ClioClient extends App {
+object ClioClient extends CommandAppWithPreCommand[CommonOptions, Command] {
+
```

Comment:
I think we lost the version-reporting. You should be able to `override val appVersion = <however we read the version>`